framework/web_app/src/services/api/ProductApiService.ts

```diff
@@ -0,0 +1,44 @@
+import type { ScrapedProductOffer } from "src/models/ScrapedProductOffer";
+import type { CreatedResponse } from "./AxiosHttpClient";
+import AxiosHttpClient from "./AxiosHttpClient";
+import { Product } from "src/models/Product";
+
+export default class ProductApiService {
+    private dapiHttpClient: AxiosHttpClient;
+    private mapiHttpClient: AxiosHttpClient;
+
+    constructor() {
+        this.dapiHttpClient = new AxiosHttpClient(5170);
+        this.mapiHttpClient = new AxiosHttpClient(5172);
+    }
+
+    // TODO: Need to decide how to deal with delete product
```

Comment:
After thinking over everything including our past discussions, i reckon once a product is created, that's it, it's in the database. If we really need to, we can include a deletion button somewhere, maybe on the product itself under the "my products" page... but that would require confirmation + what benefit does the end user get out of it? I think we're the only ones likely to think about database size and redundant information. The average user sees it all as magic voodoo. We could later on include a "delete all data" button...I reckon that would be the most useful to someone if they wanted to start fresh. If you agree we can add that to the spreadsheet.

So with that in mind, we can include a flag "is archived" on the product, which is flagged as true when a product is "unfavourited" or "unsaved". An archived product I guess we wouldn't scrape for deals on, and would be hidden from everywhere except the product search page so you can re-save it. I think it'd be a nicer end-user experience (as you previously said) to have previously scrapped deals stored in case they want to use that product again in future. Also what if someone accidentally unsaves a product? They then lose X amount of saved deals? Seems bad.

Thoughts?

Reply:
1. Agree that we just have some sort of active or saved flag and don't delete. Don't think we need to include a genuine delete for MVP. It can always be a feature request in the future for some reason.
2. Also think it is a good idea to filter out additional deals or logic for anything not active.
3. The user accidentally unsaving atm i'm not sure is a problem unless our UI is shit & you can randomly unsave products.
4. Currently we don't have the ability to unsave anyways. So i think we deal with this when we add support for the use case. **Plz resolve if happy with all.**

Reply:
Removed the comment & added a note to the dump.